### ISSUE TITLE
Support initial seed in generation

### ISSUE THREAD

**Issue by dmtrs** [wekaco/samplernn-pytorch#20]
Example implementation https://github.com/khiner/samplernn-pytorch/blob/master/generate_audio_custom_init.py#L113

**Comment by blibliki** [comment]
Generate.py -- support an extra argument  "--n_seeds" with default value 1. With N as argument script should download N random samples from dataset and run N generations for each sample. (as initial seed). File name or folder name should reflect the seed used for the generation.

 Generate.py -- support an extra argument "--explicit_seed" with default value none. With X as argument, script should download X sample of dataset and run 1 generation. File name of folder should reflect the seed (as above). If X does not exist, find X as  modulo(length of dataset).

If both   "--n_seeds" and "--explicit_seed" are given  choose  the "--explicit_seed" over  "--n_seeds"

Upload seeds and wav generations together in order to know what audio part each seed corresponds to.

**Comment by dmtrs** [comment]
When run the code from #21 to generate samples we got an unexpected result. Audio generated was in majority silent expect small chunks of audible material in the beginning and some where random in between, a similar behavior we saw in experiments we called GABBER.

@blibliki going through the original [paper](https://arxiv.org/pdf/1612.07837.pdf) particular `3.3 QUANTIFYING INFORMATION RETENTION` it states:
>For the last experiment we are interested in measuring the memory span of the model.  We trained our  model,  SampleRNN  (3-tier),  with  best  hyper-parameters  on  a  dataset  of  2  speakers  reading audio books,  one male and one female,  respectively,  with mean fundamental frequency of 125.3and 201.8Hz. Each speaker has roughly 10 hours of audio in the dataset that has been preprocesseds imilar to Blizzard. We observed that it learned to stay consistent generating samples from the same speaker without having any knowledge about the speaker ID or any other conditioning information.This effect is more apparent here in comparison to the unbalanced Onomatopoeia that sometimes mixes two different categories of sounds.

Which remind me similar behavior we had with spoken dataset we used. The sample chapter continues:

>Another experiment was conducted to test the effect of memory and study the effective memory horizon.  We inject 1 second of silence in the middle of sampling procedure in order to see if it will remember to generate from the same speaker or not.  Initially when sampling we let the model generate 2 seconds of audio as it normally do.  From 2 to 3 seconds instead of feeding back the generated sample at that timestep a silent token (zero amplitude) would be fed. From 3 to 5 seconds again we sample normally; feeding back the generated token.

Lastly...
>We  did  classification  based  on  mean  fundamental  frequency  of  speakers  for  the  first  and  last  2seconds. In 83% of samples SampleRNN generated from the same person in two separate segments.

**Comment by dmtrs** [comment]
From [other paper](https://arxiv.org/pdf/1811.06633.pdf) regarding variety it states:
> We  use  a  2-tier  SampleRNN  with  256  embedding  size,1024 dimensions, 5 layers, LSTM, 256 linear quantizationlevels, 16kHz or 32kHz sample rate, skip connections, and a 128 batch size, using weight normalization. The initial state h0 is randomized to generate more variety.

Code reference for `h0` can be found [here](https://github.com/Cortexelus/dadabots_sampleRNN/blob/master/models/two_tier/two_tier_generate32k.py#L484) where unclear of the results. 

For current implementation learning h0 can be found [here](https://github.com/wekaco/samplernn-pytorch/blob/master/model.py#L51), snippet below:
```python
        if learn_h0:
            self.h0 = torch.nn.Parameter(h0)
        else:
            self.register_buffer('h0', torch.autograd.Variable(h0))
```

and unknow what `skip connections` stand for in this implementation.
